### PR TITLE
cals 928/migrate to esm

### PR DESCRIPTION
**Pull request by joakimen**
- **feat: migrate primary project config to esm**
  

- **feat: update bundler config for esm output**
  

- **chore: cleanup package.config**
  

- **chore: pin nodejs version for dev env**
  

- **chore: clean before build**
  

- **chore: add missing dateformat lib**
  

- **chore: update chalk**
  

- **fix: replace lodash with lodash-es for esm support**
  Terrible experience working with this library during migration. In order
  to make it work with Jest, we need to replace it with the CJS version
  during test execution via module mapping in the Jest configuration.
  

- **feat: update to new yargs configuration format**
  In newer versions, yargs is no longer a singleton, and we must instead
  instantiate it directly.
  

- **fix: simplify CLI usage header due to yargs change**
  In newer versions of yargs, long usage strings (such as the previous
  logo) are subject to a few challenges:
  - they would be broken up in seemingly arbitrary places (mid-word)
  - backspaces would no longer render correctly
  - leading whitespace now gets trimmed
  
  In the following issue, the maintainers state how multiline strings were
  never an intended feature, so I reduced it to a single line.
  
  https://github.com/yargs/yargs/issues/2120
  

- **chore: update type-only imports for esm**
  

- **chore: fix formatting**

### CODE REVIEW

File: jest.config.mjs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  moduleNameMapper: {
+    "^package.json$": "<rootDir>/package.json",
+
+    // workaround: jest doesn't work well with lodash-es,
+    // so map to cjs lodash during test
+    "^lodash-es$": "lodash"
+  },
+  preset: 'ts-jest',
+  testEnvironment: "node",
+  transform: {},
```
Comment (evan-boone) [inline jest.config.mjs:11]: Kan denne linjen fjernes?

**Review by evan-boone — COMMENTED** [review]
La inn et par kommentarer! Ikke at jeg kan si at jeg er så trygg på node-økosystemet, ESM og jest, så jeg vil absolutt anbefale å teste i resources-definition :grimacing:

File: .tool-versions [hunk 2]
```diff
@@ -0,0 +1 @@
+nodejs 20.18.1
```
[the remark this answers is not in the document]
Comment (joakimen) [reply .tool-versions:1]: Jepp, kun for lokal utvikling! Pleier å ha det i versjonshåndtering i private prosjekter, og det gjøres også i Coop og Europris, pluss det støttes og holdes ved like av Renovate og. Skader ikke, men smakssak!

File: jest.config.mjs (hunk 1 above)
Comment (joakimen) [reply jest.config.mjs:11]: Tom transform var anbefalt et eller annet sted ved bruk av ESM fra Jest, husker ikke helt hvor.. Men defaultverdien er `{"\\.[jt]sx?$": "babel-jest"}` (https://jestjs.io/docs/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object), så blir nok ikke samme om vi fjerner

**Comment by joakimen** [comment]
repo-metrics: Bekrefter bruk av cals-cli ESM-pakke i repo-metrics ok. Testet webapp + collector/aggregator lambdas

**Pull request merged**